libs/blocks/gnav/gnav-breadcrumbs.js

```diff
@@ -0,0 +1,156 @@
+  const html = await res.text();
+  if (!html) return defaultItem;
+  const doc = new DOMParser().parseFromString(html, 'text/html');
+  if (!doc) return defaultItem;
+  const hideInNav = getMetadata('breadcrumbs-hide-page', doc) === 'on';
+  const hideLastPage = getMetadata('breadcrumbs-hide-last', doc) === 'on';
+  return {
+    path,
+    title: getItemTitle(doc, defaultTitle),
+    hideInNav,
+    hideLastPage,
+  };
+}
+
+async function getItems(pagePath) {
+  const items = [];
+  let path = pagePath;
+  while (path !== null) {
+    // eslint-disable-next-line no-await-in-loop
+    const item = await getItem(path);
+    if (item) items.push(item);
+    path = getParent(path);
+  }
+  items.reverse();
+  return items;
+}
+
+async function getBreadcrumbsFromUrl(pagePath) {
+  const items = await getItems(pagePath);
+  if (!items || items.length === 0) return null;
+  const ul = createTag('ul');
+  const last = items.length - 1;
+  items.forEach((item, idx) => {
+    if (item.hideInNav) return;
+    const li = createTag('li');
+    if (idx === last) {
+      if (!item.hideLastPage) {
+        li.textContent = `${item.title}`;
+        li.setAttribute('aria-current', 'page');
+      }
+    } else {
+      const a = createTag('a', { href: item.path });
+      a.textContent = `${item.title}`;
+      li.append(a);
+    }
+    ul.append(li);
+  });
+  return createTag('nav', { class: 'breadcrumbs', 'aria-label': 'Breadcrumb' }, ul);
+}
+
+function getBreadcrumbsFromPageBlock(element) {
+  const parent = element.querySelector(':scope .breadcrumbs');
+  if (!parent) return null;
+  const ul = parent.querySelector(':scope ul');
+  if (!ul) return null;
+  ul.querySelector(':scope li:last-of-type')?.setAttribute('aria-current', 'page');
+  const breadcrumbs = createTag('nav', { class: 'breadcrumbs', 'aria-label': 'Breadcrumb' }, ul);
+  parent.remove();
+  return breadcrumbs;
+}
+
+async function getBreadcrumbsFile(path) {
+  const parent = getParent(path);
+  if (!parent) return null;
+  const prefix = (parent === '/') ? '' : parent;
+  const resp = await fetch(`${prefix}/breadcrumbs.plain.html`);
+  if (resp.ok) return resp.text();
+  return getBreadcrumbsFile(parent);
+}
+
+async function getBreadcrumbsFromFile() {
+  const defaultTitle = document.location.pathname.split('/').pop();
+  const title = getItemTitle(document, defaultTitle);
+  const html = await getBreadcrumbsFile(document.location.pathname);
+  if (!html) return null;
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(html, 'text/html');
+  const breadcrumbs = getBreadcrumbsFromPageBlock(doc);
+  if (!breadcrumbs) return null;
+  // add the breadcrumbs item for the current page
+  const hideLastPage = getMetadata('breadcrumbs-hide-last') === 'on';
+  if (!hideLastPage) {
+    const ul = breadcrumbs.querySelector(':scope ul');
+    if (!ul) return null;
+    const li = createTag('li');
+    li.textContent = title;
+    li.setAttribute('aria-current', 'page');
```

Comment:
These can be added as args
```suggestion
    const li = createTag('li', { 'aria-current': 'page' }, title);
```